### PR TITLE
resolved dependency issue for sds/device/mapping resource

### PR DESCRIPTION
**Pull request by shenda1**
# Description
PR is raised to resolve dependency issues for SDS/device/mapping resources.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 80% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [ ] Acceptance Tests

image

image

image

### CODE REVIEW

File: powerflex/provider/sds_resource.go [hunk 1]
```diff
@@ -270,21 +254,53 @@ func (r *sdsResource) Read(ctx context.Context, req resource.ReadRequest, resp *
...
 
 	// Retrieve values from state
-	var state models.SdsResourceModel
 	diags = req.State.Get(ctx, &state)
 	resp.Diagnostics.Append(diags...)
 	if resp.Diagnostics.HasError() {
 		return
 	}
 
+	system, err := helper.GetFirstSystem(r.client)
```
Comment (AnikaAgiwal2711) [inline powerflex/provider/sds_resource.go:276]: Can we add this in the configure method ??

File: powerflex/provider/sds_resource.go [hunk 2]
```diff
@@ -131,9 +131,6 @@ func (r *sdsResource) Create(ctx context.Context, req resource.CreateRequest, re
 		return
 	}
 
-	// set the protection domain name in the plan so that it gets propagated to the state
-	plan.ProtectionDomainName = types.StringValue(pdm.ProtectionDomain.Name)
-
```
Comment (AnikaAgiwal2711) [inline powerflex/provider/sds_resource.go:136]: What will be the value of ProtectionDomainName if it is provided in the plan while creation ??

Comment (shenda1) [reply powerflex/provider/sds_resource.go:136]: While saving the state, the plan value gets copied for the protection domain name in the helper method.

**Comment by shenda1** [comment]
run e2e test

**Comment by shenda1** [comment]
run e2e test

File: powerflex/provider/sds_resource.go (hunk 1 above)
Comment (shenda1) [reply powerflex/provider/sds_resource.go]: done

**Review by AnikaAgiwal2711 — APPROVED** [review]

**Review by Sakshi-dell — APPROVED** [review]

**Pull request merged**